xpad.c

```diff
@@ -1106,6 +1191,48 @@ static void xpadone_process_packet(struct usb_xpad *xpad, u16 cmd, unsigned char
 			}
 		}
 
+		do_sync = true;
+
+	} else if (data[0] == 0X21) { /* The main valid packet type for GHL inputs */
```

Comment:
I think this duplicates much logic with the GIP_CMD_INPUT path - with some button remapping probably.

Can you merge this into the GIP_CMD_INPUT path? Generally, button mapping should not happen inside the driver, but rather in the level above e.g. SDL2 gamecontrollerdb.

Reply:
My data[0] = 0x21. If I want to merge into the GIP_CMD_INPUT, how would you suggest I create a value of 0x21 while maintaining logic?

Reply:
on a second thought only the handling of data[4] is somewhat similar, so I guess this can stay as is